components/script/dom/promise.rs

```diff
@@ -52,6 +52,12 @@ pub struct Promise {
     permanent_js_root: Heap<JSVal>,
 }
 
+impl PartialEq for Promise {
```

Comment:
Note: I think this a candidate for something "higher-up", maybe `DomObject`?

Reply:
Opened #33191.